src/SIL.XForge.Scripture/ClientApp/src/app/join/join.component.ts

```diff
@@ -45,40 +65,121 @@ export class JoinComponent extends DataLoadingComponent {
       if (this.authService.currentUserId == null) {
         this.i18nService.setLocale(joining.locale, this.authService);
       }
-      this.checkLinkSharing(joining.shareKey);
+      this.checkShareKey(joining.shareKey);
     });
-    const showOfflineMessage$ = combineLatest([joining$, this.pwaService.onlineStatus$]).pipe(
-      filter(([_, isOnline]) => !isOnline)
-    );
-    this.subscribe(showOfflineMessage$, () => this.showOfflineMessage());
+    this.subscribe(this.pwaService.onlineStatus$, () => this.updateOfflineJoiningStatus());
   }
 
-  private async checkLinkSharing(shareKey: string): Promise<void> {
-    this.loadingStarted();
+  get isFormEnabled(): boolean {
+    return this.status === 'input';
+  }
+
+  get inviteText(): string {
+    if (this.joiningResponse == null) {
+      return '';
+    }
+    return this.i18nService.translateAndInsertTags('join.invited_to_join', {
+      projectName: this.joiningResponse.projectName
+    });
+  }
+
+  get isJoining(): boolean {
+    return this.status === 'joining';
+  }
+
+  get isOnline(): boolean {
+    return this.pwaService.isOnline;
+  }
+
+  async joinProject(): Promise<void> {
+    if (!this.name.valid || this.name.value == null || this.joiningResponse == null) {
+      return;
+    }
+    this.status = 'joining';
+    try {
+      this.name.disable();
+      await this.anonymousService.generateAccount(
+        this.joiningResponse.shareKey,
+        this.name.value,
+        this.i18nService.localeCode
+      );
+      await this.authService.tryTransparentAuthentication();
+      if (await this.authService.isLoggedIn) {
+        await this.joinWithShareKey(this.joiningResponse.shareKey);
+      } else {
+        this.reportingService.silentError('Unable to login after generating transparent user account');
+        await this.dialogService.message('join.error_occurred_login', 'error_messages.try_again');
+        this.name.enable();
+      }
+    } catch (e) {
+      await this.informInvalidShareLinkAndRedirect();
+    }
+    this.status = 'input';
+  }
+
+  logIn(): void {
+    if (this.status != null) {
```

Comment:
When I test the code, status is not null, so clicking the log in link does not work for me?